freqtrade/strategy/interface.py

```diff
@@ -19,30 +20,80 @@ class IStrategy(ABC):
         ticker_interval -> str: value of the ticker interval to use for the strategy
     """
 
+    # associated minimal roi
     minimal_roi: Dict
+
+    # associated stoploss
     stoploss: float
+
+    # associated ticker interval
     ticker_interval: str
 
-    @abstractmethod
+    # configuration used, just in case the strategy want's to use it for something
+    config: dict = {}
+
     def populate_indicators(self, dataframe: DataFrame) -> DataFrame:
         """
         Populate indicators that will be used in the Buy and Sell strategy
         :param dataframe: Raw data from the exchange and parsed by parse_ticker_dataframe()
         :return: a Dataframe with all mandatory indicators for the strategies
         """
+        warnings.warn("deprecated - please replace this method with advise_indicators!", DeprecationWarning)
+        return dataframe
 
-    @abstractmethod
     def populate_buy_trend(self, dataframe: DataFrame) -> DataFrame:
         """
         Based on TA indicators, populates the buy signal for the given dataframe
         :param dataframe: DataFrame
         :return: DataFrame with buy column
         """
+        warnings.warn("deprecated - please replace this method with advise_buy!", DeprecationWarning)
+        dataframe.loc[
```

Comment:
Plase format this to one line: `dataframe.loc[(), 'buy'] = 0` to save lines of code, same for populate_sell_trend.